examples/htool_payload.c

```diff
@@ -68,3 +76,69 @@ int htool_payload_status() {
   }
   return 0;
 }
+
+int htool_payload_info(const struct htool_invocation* inv) {
+  struct libhoth_device* dev = htool_libhoth_device();
+  if (!dev) {
+    return -1;
+  }
+
+  const char* image_file;
+  if (htool_get_param_string(inv, "source-file", &image_file)) {
+    return -1;
+  }
+
+  int fd = open(image_file, O_RDONLY, 0);
+  if (fd == -1) {
+    fprintf(stderr, "Error opening file %s: %s\n", image_file, strerror(errno));
+    return -1;
+  }
+  struct stat statbuf;
+  if (fstat(fd, &statbuf)) {
+    fprintf(stderr, "fstat error: %s\n", strerror(errno));
+    goto cleanup;
+  }
+  if (statbuf.st_size > SIZE_MAX) {
+    fprintf(stderr, "file too large\n");
+    goto cleanup;
+  }
+
+  uint8_t* image = mmap(NULL, statbuf.st_size, PROT_READ, MAP_PRIVATE, fd, 0);
+  if (image == MAP_FAILED) {
+    fprintf(stderr, "mmap error: %s\n", strerror(errno));
+    goto cleanup;
+  }
+
+  struct payload_info info;
+  bool res = libhoth_payload_info(image, statbuf.st_size, &info);
+  if (!res) {
```

Comment:
Since you're not using res, you can simplify to 
```suggestion
  if (!libhoth_payload_info(image, statbuf.st_size, &info)) {
```